File: src/main/java/org/droidmate/analyzer/evaluation/InitialExplStrategy.java
package org.droidmate.analyzer.evaluation;

import org.droidmate.analyzer.exploration.IExplorationResult;

/**
 * Validation for the initial exploration, just check if it crashed or not
 */
public class InitialExplStrategy implements IEvaluationStrategy {
    InitialExplStrategy(){
        super();
    }

    @Override
    public boolean isValid(IExplorationResult result) {
        return (result != null) && !result.hasCrashed();
    }

    @Override
    public double getDissimilarity(IExplorationResult result) {
        return 0;
    }
}

File: src/main/kotlin/org/droidmate/analyzer/evaluation/InitialExplStrategy.kt
package org.droidmate.analyzer.evaluation

import org.droidmate.analyzer.exploration.IExplorationResult

/**
 * Validation for the initial exploration, just check if it crashed or not
 */
class InitialExplStrategy internal constructor() : IEvaluationStrategy {

    override fun isValid(result: IExplorationResult): Boolean {
        return !result.hasCrashed()
    }

    override fun getDissimilarity(result: IExplorationResult): Double {
        return 0.0
    }
}
